File: sdks/csharp/rest/src/Brume.Rest/Types/WebhookDeliveryItem.cs
using Brume.Rest.Core;
using global::System.Text.Json;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

[Serializable]
public record WebhookDeliveryItem : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("attempts")]
    public required int Attempts { get; set; }

    /// <summary>
    /// RFC-3339 timestamp of delivery creation.
    /// </summary>
    [JsonPropertyName("created_at")]
    public required string CreatedAt { get; set; }

    [JsonPropertyName("event_type")]
    public required string EventType { get; set; }

    [JsonPropertyName("id")]
    public required string Id { get; set; }

    /// <summary>
    /// RFC-3339 timestamp of the most recent attempt, or `null`.
    /// </summary>
    [JsonPropertyName("last_attempt_at")]
    public string? LastAttemptAt { get; set; }

    /// <summary>
    /// HTTP status code of the most recent attempt, or `null`.
    /// </summary>
    [JsonPropertyName("response_status")]
    public int? ResponseStatus { get; set; }

    /// <summary>
    /// `"pending" | "delivered" | "failed"`.
    /// </summary>
    [JsonPropertyName("status")]
    public required string Status { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/rest/src/Types/WebhookDeliveryItem.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;

class WebhookDeliveryItem extends JsonSerializableType
{
    /**
     * @var int $attempts
     */
    #[JsonProperty('attempts')]
    public int $attempts;

    /**
     * @var string $createdAt RFC-3339 timestamp of delivery creation.
     */
    #[JsonProperty('created_at')]
    public string $createdAt;

    /**
     * @var string $eventType
     */
    #[JsonProperty('event_type')]
    public string $eventType;

    /**
     * @var string $id
     */
    #[JsonProperty('id')]
    public string $id;

    /**
     * @var ?string $lastAttemptAt RFC-3339 timestamp of the most recent attempt, or `null`.
     */
    #[JsonProperty('last_attempt_at')]
    public ?string $lastAttemptAt;

    /**
     * @var ?int $responseStatus HTTP status code of the most recent attempt, or `null`.
     */
    #[JsonProperty('response_status')]
    public ?int $responseStatus;

    /**
     * @var string $status `"pending" | "delivered" | "failed"`.
     */
    #[JsonProperty('status')]
    public string $status;

    /**
     * @param array{
     *   attempts: int,
     *   createdAt: string,
     *   eventType: string,
     *   id: string,
     *   status: string,
     *   lastAttemptAt?: ?string,
     *   responseStatus?: ?int,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->attempts = $values['attempts'];
        $this->createdAt = $values['createdAt'];
        $this->eventType = $values['eventType'];
        $this->id = $values['id'];
        $this->lastAttemptAt = $values['lastAttemptAt'] ?? null;
        $this->responseStatus = $values['responseStatus'] ?? null;
        $this->status = $values['status'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
